code/lib/cli/src/automigrate/fixes/bare-mdx-stories-glob.test.ts

```diff
@@ -0,0 +1,205 @@
+    require('fs-extra').__setMockFiles({
+      [path.join('.storybook', 'main.js')]: `module.exports = ${JSON.stringify(main)};`,
+    });
+  } else {
+    // eslint-disable-next-line global-require
+    require('fs-extra').__setMockFiles({});
+  }
+  const packageManager = {
+    retrievePackageJson: () => ({ dependencies: {}, devDependencies: {}, ...packageJson }),
+  } as JsPackageManager;
+
+  return bareMdxStoriesGlob.check({ packageManager });
+};
+
+describe('bare-mdx fix', () => {
+  describe('should no-op', () => {
+    it('in SB < v7.0.0', async () => {
+      const packageJson = {
+        dependencies: { '@storybook/react': '^6.2.0' },
+      };
+      const main = { stories: ['../**/*.stories.mdx'] };
+      await expect(checkBareMdxStoriesGlob({ packageJson, main })).resolves.toBeFalsy();
+    });
+
+    describe('in SB >= v7.0.0', () => {
+      it('without main', async () => {
+        const packageJson = {
+          dependencies: { '@storybook/react': '^7.0.0' },
+        };
+        await expect(checkBareMdxStoriesGlob({ packageJson })).rejects.toThrow();
+      });
+
+      it('without stories field in main', async () => {
+        const packageJson = {
+          dependencies: { '@storybook/react': '^7.0.0' },
+        };
+        const main = {};
+        await expect(checkBareMdxStoriesGlob({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with variable references in stories field', async () => {
+        // eslint-disable-next-line global-require
+        require('fs-extra').__setMockFiles({
+          [path.join('.storybook', 'main.js')]: `
+          const globVar = '../**/*.stories.mdx';
+          module.exports = { stories: [globVar] };`,
+        });
+
+        const packageManager = {
+          retrievePackageJson: () => ({
+            dependencies: { '@storybook/react': '^7.0.0' },
+            devDependencies: {},
+          }),
+        } as unknown as JsPackageManager;
+
+        await expect(bareMdxStoriesGlob.check({ packageManager })).rejects.toThrow();
+      });
+
+      it('without .stories.mdx in globs', async () => {
+        const packageJson = {
+          dependencies: { '@storybook/react': '^7.0.0' },
+        };
+        const main = {
+          stories: [
+            '../src/**/*.stories.@(js|jsx|ts|tsx)',
+            { directory: '../**', files: '*.stories.@(js|jsx|ts|tsx)' },
+            { directory: '../**' },
+          ],
+        };
+        await expect(checkBareMdxStoriesGlob({ packageJson, main })).resolves.toBeFalsy();
+      });
+    });
+  });
+  describe('should fix', () => {
+    it.each([
+      {
+        existingStoriesEntries: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+        expectedStoriesEntries: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+      },
+      {
+        existingStoriesEntries: ['../src/**/*.stories.*'],
+        expectedStoriesEntries: ['../src/**/*.stories.*', '../src/**/*.mdx'],
```

Comment:
I think it would be better if we didn't create two specifiers (similar with the `files:` ones below) because that can change the order of entries (everything matching the first specifier first, then the second).

I think in this case we could make it this
```suggestion
        existingStoriesEntries: ['../src/**/*.stories.*'],
        expectedStoriesEntries: ['../src/**/*.@(mdx|stories.*)'],
```

See https://github.com/storybookjs/storybook/blob/80e43ad75d436230309ce798d9c3d594357752c0/code/lib/core-common/src/utils/normalize-stories.ts#L11

Reply:
I see, hadn't thought about the order getting changed. I'll work on it.